### DISCUSSION TITLE
How to maintain scroll position when underlying data changes?

### DISCUSSION THREAD

**Discussion by clintharris** [petyosi/react-virtuoso#880] [Q&A]
What's the best way to maintain the current scroll position of the list when the underlying `data` array changes (assuming the currently visible items are still available in both the "old" and "new" data arrays)?

I'm using `<Virtuoso>` to render a list in which the array `data` changes when the user scrolls to certain positions in the list. In other words, the current implementation is trying to use pagination in which only a single page of data is loaded at a time--and that page changes as you scroll (vs. using virtuoso's "load more" or "prepend" handlers to add more items to the `data` array that keeps growing).

Is there a `Virtuoso` property I can use to maintain the list position between `data` array changes?

**Comment by arjendevos** [comment]
Same problem here

**Comment by everduo** [comment]
firstItemIndex
